torchgeo/datasets/nlcd.py

```diff
@@ -0,0 +1,310 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""NLCD dataset."""
+
+import os
+from typing import Any, Callable, Dict, List, Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from rasterio.crs import CRS
+
+from .geo import RasterDataset
+from .utils import BoundingBox, download_url, extract_archive
+
+
+class NLCD(RasterDataset):
+    """National Land Cover Database (NLCD) dataset.
+
+    The `NLCD dataset
+    <https://www.usgs.gov/centers/eros/science/national-land-cover-database>`_
+    is a land cover product that covers the United States and Puerto Rico. It is
+    a joint effort between the United States Geological Survey
+    (`USGS <https://www.usgs.gov/>`_) and the Multi-Resolution Land Characteristics
+    Consortium (`MRLC <https://www.mrlc.gov/>`_) which released the first product
+    in 2001 with new updates every five years since then.
+
+    The dataset contains the following 15 classes:
+
+    #. Open Water
+    #. Pereniial Ice/Snow
+    #. Developed, Open Space
+    #. Developed, Low Intensity
+    #. Developed, Medium Intensity
+    #. Developed, High Intensity
+    #. Barren Land (Rock/Sand/Clay)
+    #. Deciduous Forest
+    #. Evergreen Forest
+    #. Mixed Forest
+    #. Shruf/Scrub
+    #. Grassland/Herbaceous
+    #. Pasture/Hay
+    #. Cultivated Crops
+    #. Woody Wetlands
+    #. Emergent Herbaceous Wetlands
+
+    Detailed descriptions of the classes can be found
+    `on this website <https://www.mrlc.gov/data/legends/national-land-cover-database-class-legend-and-description>`_.
```

Comment:
If you wanted to make "here" work, you would have to replace the trailing _ with __. I still hate rST...